packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -785,6 +785,10 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
   }
 
   void _dragEnd(_DragInfo item) {
+    // No changes required if last child is being inserted into the last position.
+    if (_insertIndex! + 1 == _items.length) {
+      return;
+    }
     setState(() {
       if (_insertIndex == item.index) {
         _finalDropPosition = _itemOffsetAt(_insertIndex! + (_reverse ? 1 : 0));
```

Comment:
Would it be better to clamp the value here to be within bounds? Does the list still need to call set state to complete a position change or an animation?

Reply:
If we'd want to clamp, we'd have to clamp only for `reversed: true` condition. I'd say to keep it this way. For animation to complete, `setState` is required as `_finalDropPosition` gets changed to reflect new drop offset.

Reply:
I took readings of `Offset`s of RenderObject both using flutter-framework's code and the example app code. Turns out, there's a difference of 56px between them, device's offset reading < flutter-code's offset reading.

Got 610.<exact-decimal> from example app's code;
Got 666.<exact-decimal> from flutter code;

What could this difference be? I was able to animate smoothly by hardcoding this offset, but I'd suppose it'll vary

Reply:
Update: Fixed the animation, refer code for doc. Suggest if tests are till required for this...